bcdaworker/worker/worker.go

```diff
@@ -149,11 +164,30 @@ func (w *worker) ProcessJob(ctx context.Context, job models.Job, jobArgs models.
 	return nil
 }
 
+func moveFiles(tempDir string, stagingDir string) error {
+	// Open the input file
+	files, err := os.ReadDir(tempDir)
+	if err != nil {
+		err = errors.Wrap(err, "Error reading from the staging directory for files for Job")
+		return err
+	}
+	for _, f := range files {
+		oldPath := fmt.Sprintf("%s/%s", tempDir, f.Name())
+		newPath := fmt.Sprintf("%s/%s", stagingDir, f.Name())
+		err := os.Rename(oldPath, newPath) //#nosec G304
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+
+}
+
 // writeBBDataToFile sends requests to BlueButton and writes the results to ndjson files.
 // A list of JobKeys are returned, containing the names of files that were created.
 // Filesnames can be "blank.ndjson", "<uuid>.ndjson", or "<uuid>-error.ndjson".
 func writeBBDataToFile(ctx context.Context, r repository.Repository, bb client.APIClient,
-	cmsID string, jobArgs models.JobEnqueueArgs) (jobKeys []models.JobKey, err error) {
+	cmsID string, jobArgs models.JobEnqueueArgs, tmpDir string) (jobKeys []models.JobKey, err error) {
```

Comment:
Changed the method signature to allow for passing of the directory rather than expecting it on the basis of the job ID. Of note, the []models.JobKey already had the job ID, so nothing of value is irreversibly lost here.